util/util.go

```diff
@@ -122,19 +124,27 @@ func GetServiceIDByPostmanCollectionID(c rest.FrontClient, collectionID string)
 
 		// Normalize collectionID.
 		svcCollectionID := strings.ToLower(svc.PostmanMetaData.CollectionID)
+		svcEnv := strings.ToUpper(svc.PostmanMetaData.Environment)
 
-		if strings.EqualFold(collectionID, svcCollectionID) {
+		if strings.EqualFold(collectionID, svcCollectionID) && strings.EqualFold(svcEnv, env) {
 			result = svc.ID
-			postmanCollectionIDCache.Set(svcCollectionID, svc.ID, cache.DefaultExpiration)
+			postmanCollectionIDCache.Set(svcEnv+"|"+svcCollectionID, svc.ID, cache.DefaultExpiration)
 		}
 	}
 
 	if (result != akid.ServiceID{}) {
 		printer.Stderr.Debugf("Postman collectionID %q is %q\n", collectionID, result)
 		return result, nil
 	}
-	telemetry.Failure("Unknown postman collectionID")
-	return akid.ServiceID{}, errors.Errorf("cannot determine project ID for Postman collectionID %s", collectionID)
+
+	printer.Debugf("Found no service for given collectionID: %s, creating a new service", collectionID)
+	// Create service for given postman collectionID
+	service, err := c.CreateService(ctx, "Postman-"+randomName(), collectionID, env)
+	if err != nil {
+		return akid.ServiceID{}, errors.Wrap(err, fmt.Sprintf("failed to create or get service for given collectionID: %s", collectionID))
```

Comment:
What is the end-to-end user experience like when this happens?  Should we look for this error higher up and provide some more guidance, or emit a more detailed message here?

The minimum we owe a user is: what do we want them to do?  Contact Postman?  (ugh... I haven't yet thought about brining up that subject.)  Try again later?

Can we distinguish permission problems from other sources of error?  We should be able to provide a good message when the problem is "you don't have permission to edit that collection".